### PR TITLE
ARROW-4537: [CI] Suppress shell warning on travis-ci

### PR DESCRIPTION
**Pull request by mrkn**
Enclose optional shell variable names by double quotations to suppress shell warnings that is like below:

```
+'[' == 1 ']'
/home/travis/build/apache/arrow/ci/travis_before_script_cpp.sh: line 81: [: ==: unary operator expected
```

### CODE REVIEW

File: ci/travis_before_script_cpp.sh [hunk 1]
```diff
@@ -36,7 +36,7 @@ while true; do
     esac
 done
 
-if [ "$only_library_mode" == "no" ]; then
+if [ $only_library_mode == "no" ]; then
```
Comment (xhochy) [inline ci/travis_before_script_cpp.sh]: Shouldn't the quotes stay here as-is?

**Review by kou — APPROVED** [review]
+1

**Review by xhochy — CHANGES_REQUESTED** [review]
+1, thanks!

Can you address the one line where you did the reverse?

File: ci/travis_before_script_cpp.sh (hunk 1 above)
Comment (mrkn) [reply ci/travis_before_script_cpp.sh]: It's my mistake.  I reverted this change.  Thanks for catching this.

**Review by xhochy — APPROVED** [review]

**Pull request closed without merging**